enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,244 @@
+---
+title: automatic-partition-of-disks-in-Local-Storage-Operator
+authors:
+  - "@rohan47"
+reviewers:
+  - "@hekumar"
+  - "@sapillai"
+  - "@leseb"
+  - "@travisn"
+  - "@rohantmp"
+  - "@jarrpa"
+approvers:
+  - "@hekumar"
+creation-date: 2020-09-25
+last-updated: 2020-09-25
+status:
+---
+
+# Automatic partition of Disks in the Local-Storage-Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This is a proposal to enable
+- automated partition of local devices via the `LocalVolumeSet` CR
+
+This CRD will be implemented in the LocalStorageOperator(LSO)
+
+## Motivation
+
+For performance reasons the user will want to use partitions.
+On NVMe devices, some of the devices are capable of delevering very high IOPS per device. An OSD tends to bottleneck long before it reaches anywhere close to the device IOPS limit.
+So with single OSD per NVMe device, there will be a big gap between measured or advertised capabilities of these devices and what you can get with Ceph/OCS on top of them.
+
+### Goals
+
+- Partition the selected devices based on the partition policy specified by the user.
+- If a LocalVolumeSet has partitioning policy, it will create partitions on all disks.
+
+### Non-Goals
+
+- If anything changes later such as the parition policy changes or the device is resized, we won't go add or change any partitions.
+- Policy for wiping device to clean the partitions
+- Mixed method of PV creation is not supported. i.e., using some devices entirely and some with partitoins in a single LocalVolumeSet is not supported.
+
+### Risks and Mitigations
```

Comment:
Not sure I understand the point of the risks section. You are pointing out that we might partition the wrong disk if the user gives the wrong device filter? This risk doesn't seem any different from risks of the LocalVolumeSet already has, so maybe we can just refer back to that doc if it has a risks section.